handlers/api.go

```diff
@@ -1766,6 +1775,39 @@ func ApiValidatorIncomeDetailsHistory(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getIncomeDetailsHistoryQueryParameters(q url.Values) (uint64, uint64, error) {
+	onChainLatestEpoch := services.LatestEpoch()
```

Comment:
This needs to be `LatestFinalizedEpoch`.

Before your changes the results were from the last 102 epochs starting from the `LatestEpoch`.
Since most of the time 2 are not yet finalized that resulted in the desired 100 epoch output.
However the correct way would be to use `LatestFinalizedEpoch` and a `limit` of 100